libminifi/src/Connection.cpp

```diff
@@ -157,15 +149,15 @@ void Connection::put(const std::shared_ptr<core::FlowFile>& flow) {
 
 void Connection::multiPut(std::vector<std::shared_ptr<core::FlowFile>>& flows) {
   {
-    std::lock_guard<std::mutex> lock(mutex_);
+    auto lockedQueue = queue_.lock();
```

Comment:
Instead of this, I would create an interface on the queue that performs batch insertion. This piece of API could be better tested as well.